`cpanm Term::ReadLine::Gnu` fails under docker

**Describe the bug**
Can't install `cpanm Term::ReadLine::Gnu` under docker (`TERM` is not set).

**To Reproduce**
Steps to reproduce the behavior:

`Dockerfile`:

```dockerfile
FROM alpine:3.16
RUN set -x \
    && apk add perl perl-app-cpanminus ncurses-dev readline-dev perl-dev build-base \
    && cpanm Term::ReadLine::Gnu || { r=$?; cat /root/.cpanm/work/*/build.log; exit $r; }
```

```
$ docker build -t i .
...
+ cpanm Term::ReadLine::Gnu
--> Working on Term::ReadLine::Gnu
Fetching https://www.cpan.org/authors/id/H/HA/HAYASHI/Term-ReadLine-Gnu-1.42.tar.gz ... OK
Configuring Term-ReadLine-Gnu-1.42 ... OK
Building and testing Term-ReadLine-Gnu-1.42 ... ! Installing Term::ReadLine::Gnu failed. See /root/.cpanm/work/1662628609.10/build.log for details. Retry with --force to force install it.
FAIL
+ r=1
+ cat /root/.cpanm/work/1662628609.10/build.log
cpanm (App::cpanminus) 1.7046 on perl 5.034001 built for x86_64-linux-thread-multi
Work directory is /root/.cpanm/work/1662628609.10
You have make /usr/bin/make
You have /usr/bin/wget
You have /bin/tar: tar (busybox) 1.35.0
You have /usr/bin/unzip
Searching Term::ReadLine::Gnu () on cpanmetadb ...
--> Working on Term::ReadLine::Gnu
Fetching https://www.cpan.org/authors/id/H/HA/HAYASHI/Term-ReadLine-Gnu-1.42.tar.gz
-> OK
Unpacking Term-ReadLine-Gnu-1.42.tar.gz
Entering Term-ReadLine-Gnu-1.42
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.62)
Configuring Term-ReadLine-Gnu-1.42
Running Makefile.PL
Warning (mostly harmless): No library found for -ltermcap
Warning (mostly harmless): No library found for -ltinfo
<<<It seems that you have the GNU Readline Library version 8.1.>>>
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Term::ReadLine::Gnu
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.62)
Building and testing Term-ReadLine-Gnu-1.42
cp Gnu.pm blib/lib/Term/ReadLine/Gnu.pm
cp Gnu/XS.pm blib/lib/Term/ReadLine/Gnu/XS.pm
AutoSplitting blib/lib/Term/ReadLine/Gnu/XS.pm (blib/lib/auto/Term/ReadLine/Gnu/XS)
blib/lib/Term/ReadLine/Gnu/XS.pm: some names are not unique when truncated to 8 characters:
 directory blib/lib/auto/Term/ReadLine/Gnu/XS:
  rl_bind_key.al, rl_bind_key_if_unbound.al, rl_bind_keyseq.al, rl_bind_keyseq_if_unbound.al truncate to rl_bind_
  rl_set_keymap.al, rl_set_key.al truncate to rl_set_k
Running Mkbootstrap for Gnu ()
chmod 644 "Gnu.bs"
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Gnu.bs blib/arch/auto/Term/ReadLine/Gnu/Gnu.bs 644
"/usr/bin/perl" "/usr/share/perl5/core_perl/ExtUtils/xsubpp"  -typemap '/usr/share/perl5/core_perl/ExtUtils/typemap' -typemap '/root/.cpanm/work/1662628609.10/Term-ReadLine-Gnu-1.42/typemap'  Gnu.xs > Gnu.xsc
mv Gnu.xsc Gnu.c
cc -c   -D_REENTRANT -D_GNU_SOURCE -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Os -fomit-frame-pointer   -DVERSION=\"1.42\" -DXS_VERSION=\"1.42\" -fPIC "-I/usr/lib/perl5/core_perl/CORE"  -DTRG_READLINE_VERSION=0x0801 Gnu.c
rm -f blib/arch/auto/Term/ReadLine/Gnu/Gnu.so
LD_RUN_PATH="/usr/lib" cc  -shared -Os -fomit-frame-pointer -L/usr/local/lib -fstack-protector-strong  Gnu.o  -o blib/arch/auto/Term/ReadLine/Gnu/Gnu.so  \
   -lreadline -lncurses   \
  
chmod 755 blib/arch/auto/Term/ReadLine/Gnu/Gnu.so
cp eg/perlsh blib/script/perlsh
"/usr/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/perlsh
Manifying 1 pod document
Manifying 1 pod document
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Gnu.bs blib/arch/auto/Term/ReadLine/Gnu/Gnu.bs 644
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
Use of uninitialized value $Term::ReadLine::Gnu::VERSION in concatenation (.) or string at t/00checkver.t line 27.

#   Failed test 'An object of class 'Term::ReadLine::Stub' isa 'Term::ReadLine''
#   at t/00checkver.t line 30.
#     The object of class 'Term::ReadLine::Stub' isn't a 'Term::ReadLine'

#   Failed test ''Attribs' isa 'Term::ReadLine''
#   at t/00checkver.t line 32.
#     'Attribs' isn't a 'Term::ReadLine'
Use of uninitialized value in concatenation (.) or string at t/00checkver.t line 36.
Use of uninitialized value in printf at t/00checkver.t line 36.
Use of uninitialized value $ENV{"TERM"} in concatenation (.) or string at t/00checkver.t line 37.
# Looks like you failed 2 tests of 4.
t/00checkver.t ... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/4 subtests 
t/01test_use.t ... ok
t/02test_use.t ... ok

#   Failed test 'An object of class 'Term::ReadLine::Stub' isa 'Term::ReadLine''
#   at t/callback.t line 36.
#     The object of class 'Term::ReadLine::Stub' isn't a 'Term::ReadLine'

#   Failed test ''Attribs' isa 'Term::ReadLine''
#   at t/callback.t line 38.
#     'Attribs' isn't a 'Term::ReadLine'
Use of uninitialized value in pattern match (m//) at t/callback.t line 40.
# skipped since Tk is not available.
# Looks like you failed 2 tests of 8.
t/callback.t ..... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/8 subtests 
Use of uninitialized value $Term::ReadLine::Gnu::VERSION in concatenation (.) or string at t/history.t line 34.

#   Failed test 'An object of class 'Term::ReadLine::Stub' isa 'Term::ReadLine''
#   at t/history.t line 40.
#     The object of class 'Term::ReadLine::Stub' isn't a 'Term::ReadLine'

#   Failed test '$t->ReadLine'
#   at t/history.t line 46.

#   Failed test ''Attribs' isa 'Term::ReadLine''
#   at t/history.t line 53.
#     'Attribs' isn't a 'Term::ReadLine'
Use of uninitialized value in pattern match (m//) at t/history.t line 55.
Can't locate object method "using_history" via package "Term::ReadLine::Stub" at t/history.t line 62.
# Looks like your test exited with 255 just after 4.
t/history.t ...... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 87/88 subtests 
Bareword "ISKMAP" not allowed while "strict subs" in use at t/readline.t line 300.
Bareword "ISFUNC" not allowed while "strict subs" in use at t/readline.t line 303.
Bareword "ISMACR" not allowed while "strict subs" in use at t/readline.t line 307.
Bareword "RL_STATE_INITIALIZED" not allowed while "strict subs" in use at t/readline.t line 150.
Bareword "ISFUNC" not allowed while "strict subs" in use at t/readline.t line 220.
Bareword "ISFUNC" not allowed while "strict subs" in use at t/readline.t line 232.
Bareword "ISKMAP" not allowed while "strict subs" in use at t/readline.t line 316.
Bareword "ISMACR" not allowed while "strict subs" in use at t/readline.t line 320.
Bareword "prompt" not allowed while "strict subs" in use at t/readline.t line 657.
Bareword "prompt" not allowed while "strict subs" in use at t/readline.t line 661.
Bareword "prompt" not allowed while "strict subs" in use at t/readline.t line 663.
Bareword "prompt" not allowed while "strict subs" in use at t/readline.t line 667.
Bareword "prompt" not allowed while "strict subs" in use at t/readline.t line 669.
Bareword "ISFUNC" not allowed while "strict subs" in use at t/readline.t line 998.
Bareword "ISKMAP" not allowed while "strict subs" in use at t/readline.t line 998.
Bareword "ISMACR" not allowed while "strict subs" in use at t/readline.t line 998.
Bareword "prompt" not allowed while "strict subs" in use at t/readline.t line 1007.
Execution of t/readline.t aborted due to compilation errors.
# Looks like your test exited with 255 before it could output anything.
t/readline.t ..... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 148/148 subtests 
Use of uninitialized value $Term::ReadLine::Gnu::VERSION in concatenation (.) or string at t/utf8_binary.t line 41.

#   Failed test 'An object of class 'Term::ReadLine::Stub' isa 'Term::ReadLine''
#   at t/utf8_binary.t line 88.
#     The object of class 'Term::ReadLine::Stub' isn't a 'Term::ReadLine'

#   Failed test 'input layers after 'new''
#   at t/utf8_binary.t line 93.
#     Structures begin differing at:
#          $got->[2] = Does not exist
#     $expected->[2] = 'stdio'

#   Failed test 'output layers after 'new''
#   at t/utf8_binary.t line 96.
#     Structures begin differing at:
#          $got->[2] = Does not exist
#     $expected->[2] = 'stdio'
# Looks like you failed 3 tests of 13.
t/utf8_binary.t .. 
Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/13 subtests 
Use of uninitialized value $Term::ReadLine::Gnu::VERSION in concatenation (.) or string at t/utf8_text.t line 46.

#   Failed test 'An object of class 'Term::ReadLine::Stub' isa 'Term::ReadLine''
#   at t/utf8_text.t line 114.
#     The object of class 'Term::ReadLine::Stub' isn't a 'Term::ReadLine'
# Looks like you failed 1 test of 14.
t/utf8_text.t .... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/14 subtests 

Test Summary Report
-------------------
t/00checkver.t (Wstat: 512 Tests: 4 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
t/callback.t   (Wstat: 512 Tests: 8 Failed: 2)
  Failed tests:  2-3
  Non-zero exit status: 2
t/history.t    (Wstat: 65280 Tests: 4 Failed: 3)
  Failed tests:  2-4
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 88 tests but ran 4.
t/readline.t   (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 148 tests but ran 0.
t/utf8_binary.t (Wstat: 768 Tests: 13 Failed: 3)
  Failed tests:  6-8
  Non-zero exit status: 3
t/utf8_text.t  (Wstat: 256 Tests: 14 Failed: 1)
  Failed test:  7
  Non-zero exit status: 1
Files=8, Tests=47,  0 wallclock secs ( 0.04 usr  0.01 sys +  0.46 cusr  0.06 csys =  0.57 CPU)
Result: FAIL
Failed 6/8 test programs. 11/47 subtests failed.
make: *** [Makefile:1080: test_dynamic] Error 1
-> FAIL Installing Term::ReadLine::Gnu failed. See /root/.cpanm/work/1662628609.10/build.log for details. Retry with --force to force install it.
+ exit 1
The command '/bin/sh -c set -x     && apk add perl perl-mojolicious perl-app-cpanminus ncurses-dev readline-dev perl-dev build-base     && cpanm Term::ReadLine::Gnu || { r=$?; cat /root/.cpanm/work/*/build.log; exit $r; }' returned a non-zero code: 1
```

**Expected behavior**
It installs.

**Please complete the following information:**
 - Output of
   - perl -V
   - perl Makefile.PL verbose
   - make test TEST_VERBOSE=1
   - perl -Mblib t/00checkver.t
   - echo $TERM
 - terminal emulator which you are using
 - compiler which is used to compile the GNU Readline Library (libreadline.a) if you can know.

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -1,7 +1,7 @@
 #
 #       Makefile.PL for Term::ReadLine::Gnu
 #
-#       Copyright (c) 1996-2021 Hiroo Hayashi.  All rights reserved.
+#       Copyright (c) 1996-2022 Hiroo Hayashi.  All rights reserved.
 #               <hiroo.hayashi@computer.org>
 #
 #       This program is free software; you can redistribute it and/or
@@ -30,12 +30,11 @@ if ($ENV{AUTOMATED_TESTING} && !open(SESAMI, '/dev/tty')) {
     warn "cannot open /dev/tty\n";
     exit 0;
 }
-# test $TERM for CPAN Testers
 # https://rt.cpan.org/Ticket/Display.html?id=128806
-if ($ENV{AUTOMATED_TESTING}
-    && (!exists($ENV{TERM}) || !defined($ENV{TERM}) || $ENV{TERM} =~ /^(dumb|emacs|unknown|)$/)) {
-    warn "wrong \$TERM value: $ENV{TERM}\n";
-    exit 0;
+# https://github.com/hirooih/perl-trg/issues/11
+if (!exists($ENV{TERM}) || !defined($ENV{TERM}) || $ENV{TERM} =~ /^(dumb|emacs|unknown|)$/) {
+    warn "Warning: wrong \$TERM value, '$ENV{TERM}'. The GNU Readline Library does not work properly.\n";
+    exit 0 if $ENV{AUTOMATED_TESTING};  # exit on CPAN testers
 }
 
 push(@defs, '-DHAVE_STRING_H') if ($Config{strings} =~ m|/string.h$|);

--- a/t/00checkver.t
+++ b/t/00checkver.t
@@ -1,14 +1,14 @@
 # -*- perl -*-
 #	00check.t - check versions
 #
-#	Copyright (c) 2008-2016 Hiroo Hayashi.  All rights reserved.
+#	Copyright (c) 2008-2022 Hiroo Hayashi.  All rights reserved.
 #
 #	This program is free software; you can redistribute it and/or
 #	modify it under the same terms as Perl itself.
 
 use strict;
 use warnings;
-use Test::More tests => 4;
+use Test::More tests => 5;
 use vars qw($loaded);
 
 BEGIN {
@@ -17,13 +17,22 @@ BEGIN {
 END {
     unless ($loaded) {
 	ok(0, 'fail before loading');
-	diag "\nPlease report the output of \'perl Makefile.PL\'\n"; 
+	diag "\nPlease report the output of \'perl Makefile.PL\'\n";
     }
 }
 
 use Term::ReadLine;
 ok(1, 'load done');
 $loaded = 1;
+
+# The GNU Readline library requires $TERM to be set properly.
+# https://github.com/hirooih/perl-trg/issues/11
+if (!exists($ENV{TERM}) || !defined($ENV{TERM}) || $ENV{TERM} =~ /^(dumb|emacs|unknown|)$/) {
+    warn "wrong \$TERM value: $ENV{TERM}\n";
+    exit 1;
+}
+ok(1, '$TERM value');
+
 print "# I'm testing Term::ReadLine::Gnu version $Term::ReadLine::Gnu::VERSION\n";
 
 my $t = new Term::ReadLine 'ReadLineTest';

